rfd/0194-k8s-rbac-simplification.md

```diff
@@ -0,0 +1,210 @@
+
+The goal is to streamline the user experience and reduce the complexity of
+getting up and running on day 1 and managing permissions later on.
+
+## Why
+
+Today, the Kubernetes RBAC in Teleport is complex and can be difficult for
+users to manage.
+A proper setup requires some external Kubernetes config and matching role
+setup.
+This complexity leads to misconfigurations and security issues.
+
+Simplifying the RBAC model will help improve usability and security.
+
+### References
+
+A common issue currently is the un-intuitive result of complex rule sets and
+more critically, difficult to get setup on day 1.
+
+- @klizhentas struggling to setup a cluster following the `self-hosted` flow:
+  (internal) <https://gravitational.slack.com/archives/C03SULRAAG3/p1715394587181739>
+- Customer confusion on role resolution:
+  - <https://github.com/gravitational/teleport/issues/45475>
+  - (internal) <https://goteleport.slack.com/archives/C0582MBSMHN/p1738187497666819>
+- Unexpected `exec` grant on read-only user:
+  (internal) <https://gravitational.slack.com/archives/C32M8FP1V/p1739462454321459?thread_ts=1739462384.714419&cid=C32M8FP1V>
+
+## Goals
+
+- Make the Teleport agent authoritative
+  - Handle permissions for the users instead of relying on impersonating a
+    different user/group
+- Discourage (or even deprecate?) the use of the `kubernetes_groups` and
+  `kubernetes_users` in favor of `kubernetes_resources`
+- Clarify / Improve documentation for the various ways to setup Kubernetes in
+  Teleport.
+
+## Proposal
+
+### Background
+
+Currently, when enrolling a Kubernetes cluster and setting it up to use the
+group `system:masters`, everything works as expected. Teleport is super admin
+and yields reduced permissions to the user based on the `kubernetes_resources`
+and `kubernetes_label` fields from the Teleport role.
+
+### Impersonation
+
+To limit the number of code changes, the proposal is to default the
+impersonation to use the Teleport service account.
+
+We will need to make sure the service account has `cluster-admin` permissions.
+
+To avoid unexpected access being granted, the `kubernetes_resources` field
+should default to `null` as well instead of the current wildcard.
+
+The `kubernetes_user` should default to the Teleport user auth email.
+
+### UX
+
+#### Deprecate `namespace`
+
+We currently have the `namespace` kind mean "everything within a namespace",
+which has caused confusion with customers and resulting in unexpected access
+granted.
+
+This should be deprecated and removed. It would require a bump on the role
+model to v8 to avoid changing behavior for current customers.
+
+Upgrading to v8 would map the deprecated `namespace` field to `*` while
+swapping `name`/`namespace` as these are equivalent within a namespace:
+
+```yaml
+kind: namespace
+name: foo
+```
+
+and:
+
+```yaml
+kind: '*'
+namespace: foo
+```
+
+Note that the later also grants access to cluster-wide resources outside a
```

Comment:
This will be even more confusing than the current namespace behavior.  

When a namespace is specified, it should provide access to the resources within that namespace. For cluster-wide resources, the namespace should be left empty.